backend/README.md

```diff
@@ -53,6 +53,27 @@ can issue GraphQL requests or access GraphQL Playground in the browser.
 
 ![GraphQL Playground](../.gitbook/assets/graphql-playground.png)
 
+### Database Indices and Constraints
+
+Database indices and constraints need to be created when the database and the
+backend is running:
+
+{% tabs %}
+{% tab title="Docker" %}
+```bash
+docker-compose exec backend yarn run db:migrate init
```

Comment:
looking into this further @roschaefer ... this has nothing, it seems, to do with `Docker` vs. local setup, but production vs seed data.
It appears there is an error thrown when setting up the unique constraints on `Location` and then the rest of the 4 :point_up_2: don't get run. I ran them individually in the `neo4j` browser and got this error for `Location`, the other 4 were successful.

```
Unable to create CONSTRAINT ON ( location:Location ) ASSERT location.id IS UNIQUE:
Both Node(48981) and Node(186812) have the label `Location` and property `id` = 'place.4463223470618060'
```
Looking this `Location` up by it's id results in 4 identical `Locations` in production data, that look like
```
{
  "lng": 48.06028,
  "nameES": "Villingen-Schwenningen",
  "nameFR": "Villingen-Schwenningen",
  "nameIT": "Villingen-Schwenningen",
  "nameEN": "Villingen-Schwenningen",
  "type": "place",
  "namePT": "Villingen-Schwenningen",
  "nameDE": "Villingen-Schwenningen",
  "nameNL": "Villingen-Schwenningen",
  "name": "Villingen-Schwenningen",
  "namePL": "Villingen-Schwenningen",
  "id": "place.4463223470618060",
  "lat": 8.45861
}
```

I guess we don't really need to run `init` in production, right? should we delete these duplicate records anyways? 
Looking at our db schema in production confirms that we do no have any `CONSTRAINT` `on `Location`, so I guess we want to set that up?

Reply:
@mattwr18 yes, we want to set that up. I used `neode` to setup unique constraints and if that fails on production data, then we had one constraint missing on `production`. Maybe we can run a migration before we setup the constraints. Sounds odd but is the best solution that comes to my mind right now.

Reply:
I've added a migration that runs first... but we would need to run `yarn db:migrate up` before running `yarn db:migrate init`... otherwise the latter will fail.
Also, I changed the timestamp of the `merge_duplicate_location_nodes` so that it run before the `merge_duplicate_user_nodes`, but actually that's not important.
We have talked about adding the migrations we have already run on production data to this folder so that we have a comprehensive list of every migration we have run... should those timestamps be altered as well to appear at the top of the list and be run first?
Also, when I added the unique constraints on `Migration` node, I realized it failed as it was creating multiple nodes for the same migration.
I've updated it to `MERGE` by title, then `ON CREATE SET` the properties and changed the timestamp to `createdAt` so that we have a record of when the migration was run in production, and it's human readable.